File: backend/app/schemas/candidate.py
from pydantic import BaseModel, ConfigDict
from typing import List


class CandidateCreate(BaseModel):
    name: str 
    experience: int
    education: str 
    skills: List[str]
    desired_salary: int
    can_relocate: bool

    model_config = ConfigDict(from_attributes=True)

class CandidateResponse(BaseModel):
    id: int
    name: str 
    experience: int
    education: str 
    skills: List[str]
    desired_salary: int
    can_relocate: bool

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/candidate.ts
export interface Candidate {
  id: number;
  name: string;
  email: string;
  phone: string;
  experience: number;
  skills: string[];
  education: 'higher_profile' | 'higher_other' | 'secondary';
  desired_salary: number;
  can_relocate: boolean;
}

export interface CandidateCreate {
  name: string;
  email: string;
  phone: string;
  experience: number;
  skills: string[];
  education: 'higher_profile' | 'higher_other' | 'secondary';
  desired_salary: number;
  can_relocate: boolean;
}